src/chrome/content/code/HTTPSRules.js

```diff
@@ -311,16 +311,19 @@ const RuleWriter = {
       this.log(WARN, "TARGETS IS NULL");
     var data = this.readFromFile(file);
     if (!data) return null;
-    return this.readFromString(data, rule_store, ruleset_id);
+    return this.readFromString(data, rule_store, ruleset_id, true);
   },
 
-  readFromString: function(data, rule_store, ruleset_id) {
+  readFromString: function(data, rule_store, ruleset_id, custom) {
```

Comment:
Rather than parsing user rules before exporting them (soon to be: copying them via the message-passing API), I think it would be simpler to have the export function read the raw file contents, and send them across as a list of strings. That way the thin-shell bootstrapped XPCOM extension won't need to include all the HTTPSRules parsing code.